__tests__/api.test.js

```diff
@@ -125,3 +137,61 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("POST:201 posts comment on specific article for user that doesn't exist", () => {
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send({
+        username: "Adam",
+        body: "new comment",
+      })
+      .expect(201)
+      .then((res) => {
+        const comments = res.body.article;
+        expect(comments.length).toBe(1);
+        comments.forEach((comment) => {
+          expect(comment).toHaveProperty("comment_id");
+          expect(comment).toHaveProperty("votes");
+          expect(comment).toHaveProperty("created_at");
+          expect(comment).toHaveProperty("author", "Adam");
+          expect(comment).toHaveProperty("body", "new comment");
+          expect(comment).toHaveProperty("article_id", 2);
+        });
+      });
+  });
+  test("POST:201 posts comment on specific article for user that exists", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({
+        username: "lurker",
+        body: "new comment",
+      })
+      .expect(201)
+      .then((res) => {
+        let comments = res.body.article;
+        expect(comments.length).toBe(12);
+        comments = comments.filter((comment) => comment.author === "lurker");
+
+        comments.forEach((comment) => {
+          expect(comment).toHaveProperty("comment_id");
+          expect(comment).toHaveProperty("votes");
+          expect(comment).toHaveProperty("created_at");
+          expect(comment).toHaveProperty("author", "lurker");
+          expect(comment).toHaveProperty("body", "new comment");
+          expect(comment).toHaveProperty("article_id", 1);
+        });
+      });
+  });
+  test("POST:400 sends appropriate error message when send bad request", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({
+        body: "new comment",
+      })
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe("bad request");
+      });
+  });
+});
```

Comment:
tests needed for: 

 - valid but non existing endpoint 
 - invalid endpoint id
 - the above mention comment about invalid user